django/test/testcases.py

```diff
@@ -1044,8 +1044,17 @@ def _fixture_setup(self):
     def _fixture_teardown(self):
         if not connections_support_transactions():
             return super(TestCase, self)._fixture_teardown()
+        for db_name in reversed(self._databases_names()):
+            if self._should_check_constraints(connections[db_name]):
+                connections[db_name].check_constraints()
```

Comment:
Could adding a try/finally here prevent cascading failures across the entire test suite? I think I observed the behavior when I was testing the patch earlier.